cmd/kubelet/app/server.go

```diff
@@ -1260,6 +1260,10 @@ func RunKubelet(ctx context.Context, kubeServer *options.KubeletServer, kubeDeps
 
 	// process pods and exit.
 	if runOnce {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.LegacyNodeRunOnceMode) {
+			panic("Cannot run runonce mode when LegacyNodeRunOnceMode feature gate is disabled")
+		}
+		klog.ErrorS(nil, "RunOnce mode is deprecated,  please migrate to the podman kube subcommand.")
```

Comment:
Should kubelet be referencing podman specifically?  Not everyone uses podman.  This feels like it should be a simple error and docs can talk about options for people who really need this.

Reply:
Perhaps I can point them to the `Risks and Mitigations` section of the KEP, where those looking for alternatives can find answers.